external/anomalib/tests/helpers/dummy_dataset.py

```diff
@@ -0,0 +1,241 @@
+        num_test: int = 100,
+        img_height: int = 128,
+        img_width: int = 128,
+        max_size: int = 10,
+        train_shapes: List[str] = ["triangle", "rectangle"],
+        test_shapes: List[str] = ["hexagon", "star"],
+        dataset_path: Union[str, Path] = "./datasets/MVTec",
+        use_mvtec: bool = False,
+    ) -> None:
+        """Creates a context for Generating Dummy Dataset. Useful for wrapping test functions.
+        NOTE: for MVTec dataset it does not return a category.
+        It is adviced to use a default parameter in the function
+
+        Args:
+            num_train (int, optional): Number of training images to generate. Defaults to 1000.
+            num_test (int, optional): Number of testing images to generate per category. Defaults to 100.
+            img_height (int, optional): Height of the image. Defaults to 128.
+            img_width (int, optional): Width of the image. Defaults to 128.
+            max_size (Optional[int], optional): Maximum size of the test shapes. Defaults to 10.
+            train_shapes (List[str], optional): List of good shapes. Defaults to ["circle", "rectangle"].
+            test_shapes (List[str], optional): List of anomalous shapes. Defaults to ["triangle", "ellipse"].
+            path (Union[str, Path], optional): Path to MVTec dataset. Defaults to "./datasets/MVTec".
+            use_mvtec (bool, optional): Use MVTec dataset or dummy dataset. Defaults to False.
+
+        Example:
+            >>> @TestDataset
+            >>> def test_some_function(path, category="leather"):
+            >>>     do something
+        """
+        self.num_train = num_train
+        self.num_test = num_test
+        self.img_height = img_height
+        self.img_width = img_width
+        self.max_size = max_size
+        self.train_shapes = train_shapes
+        self.test_shapes = test_shapes
+        self.dataset_path = dataset_path
+        self.use_mvtec = use_mvtec
+
+    def __call__(self, func):
+        @wraps(func)
+        def inner(*args, **kwds):
+            # If true, will use MVTech dataset for testing.
+            # Useful for nightly builds
+            if self.use_mvtec:
+                return func(*args, dataset_path=self.dataset_path, **kwds)
+            else:
+                with GeneratedDummyDataset(
+                    num_train=self.num_train,
+                    num_test=self.num_test,
+                    img_height=self.img_height,
+                    img_width=self.img_width,
+                    train_shapes=self.train_shapes,
+                    test_shapes=self.test_shapes,
+                    max_size=self.max_size,
+                ) as dataset_path:
+                    kwds["category"] = "shapes"
+                    return func(*args, dataset_path=dataset_path, **kwds)
+
+        return inner
+
+
+class GeneratedDummyDataset(ContextDecorator):
+    """Context for generating dummy shapes dataset.
+    Example:
```

Comment:
there should be a newline before `Example`. I know we're not gonna run sphinx, but would be nice in terms of consistency.

Reply:
And same for the other stuff down below